test/test_invalid_headers.py

```diff
@@ -239,22 +241,24 @@ class TestFilter(object):
     ]
 
     hdr_validation_combos = [
-        h2.utilities.HeaderValidationFlags(is_client, is_trailer)
-        for is_client, is_trailer in [
-            (True, True),
-            (True, False),
-            (False, True),
-            (False, False)
-        ]
+        h2.utilities.HeaderValidationFlags(
+            is_client, is_trailer, is_response_header
+        )
+        for is_client, is_trailer, is_response_header in (
+            itertools.product([True, False], repeat=3)
+        )
     ]
 
-    hdr_validation_no_trailers = [
-        h2.utilities.HeaderValidationFlags(is_client, is_trailer)
-        for is_client, is_trailer in [
-            (True, False),
-            (False, False)
-        ]
-    ]
+    hdr_validation_response_headers = list(
+        filter(lambda flags: flags.is_response_header, hdr_validation_combos)
+    )
```

Comment:
(Opinion) I’d prefer this as a list comprehension, as I think that’s easier to see at a glance what the condition is, i.e.,

```python
hdr_validation_response_headers = [
    flags for flags in hdr_validation_combos
    if flags.is_response_header
]
```

and

```python
hdr_validation_response_headers = [
    flags for flags in hdr_validation_combos
    if not (flags.is_trailer or flags.is_response_header)
]
```

but if @Lukasa is happy with `list(filter(lambda(...)))`, I’ll shut up about it.